hail/src/main/scala/is/hail/expr/types/virtual/TNDArray.scala

```diff
@@ -20,7 +20,7 @@ object TNDArray {
 
 final case class TNDArray(elementType: Type, nDimsBase: NatBase, override val required: Boolean = false) extends Type {
   lazy val nDims: Int = {
-    assert(nDimsBase.isInstanceOf[Nat], "Missing concrete number of dimensions.")
+    assert(nDimsBase.isInstanceOf[Nat], s"Missing concrete number of dimensions. nDimsBase was $nDimsBase")
```

Comment:
Was this just debugging, or do you think it's important to print the message? The `asInstanceOf` will already raise an exception if it fails. If you think we want the message, this should be written
```
if (!nDimsBase.isInstanceOf[Nat]) throw ...
nDimsBase.asInstanceOf[Nat].n
```
to prevent `assert` allocating a closure.

Reply:
Nah just debugging. I'll remove.